docs/testing.md

```diff
@@ -0,0 +1,179 @@
+
+Theoretically, you could go so far as to move all logic and data fetching out of your components. This way your components would be solely dedicated to rendering, and your state would be independent on your components - your app’s logic would work without any React components at all!
+
+After writing testable code, it’s time to write some actual tests! (Or if you do test-driven development, you actually write tests first!).
+
+React Native apps come with [Jest](https://jestjs.io) out of the box, ready to use. Jest is a popular JavaScript Testing Framework also started at Facebook. You can use it to write all types of tests featured in this guide. Let’s get started!
+
+### Structuring Tests
+
+Your tests should be short and ideally test only one thing. Let's start with an example unit test:
+
+```js
+it('given a date in the past, colorForDueDate() returns red', () => {
+    expect(colorForDueDate('2000-10-20'))).toBe('red');
+});
+```
+
+The test is described by the string passed to the [`it`](https://jestjs.io/docs/en/api#testname-fn-timeout) function - take a good care writing the description so that it’s clear what is being tested. Do your best to cover the following:
+
+1. **Given** - some precondition
+2. **When** - some action executed by the function that you’re testing
+3. **Then** - the expected outcome
+
+This is also known as AAA (Arrange, Act, Assert).
+
+Jest offers [`describe`](https://jestjs.io/docs/en/api#describename-fn) function to help structure your tests. Use `describe` to group together all tests that belong to one functionality. Describes can be nested, if you need that. Other functions you'll use are [`beforeEach`](https://jestjs.io/docs/en/api#beforeeachfn-timeout) or [`beforeAll`](https://jestjs.io/docs/en/api#beforeallfn-timeout) that you can use for setting up the objects you're testing. Read more in the [Jest api reference](https://jestjs.io/docs/en/api).
+
+If your test has many steps or many expectations, you probably want to split it into multiple smaller ones. Also, ensure that your tests are completely independent of one another. Each test in your suite must be executable on its own without eg. first running some other test. Conversely, if you run all your tests together, the first test must not influence the output of the second one.
+
+Lastly, as developers we like when our code works great and doesn't crash. With tests, this is often the opposite - think of a failing test as of a _good thing_! When a test is failing, it means something is not right and it gives us an opportunity to fix the problem before it impacts the users.
+
+## Unit tests
+
+The lowest level at which you can test is unit testing. Unit tests cover the smallest parts of code, like individual functions or classes.
```

Comment:
Is "The lowest level at which you can test is unit testing" a phrase that will make sense? Is this sentence necessary or is the following sentence enough?

Reply:
I think just the second sentence is enough, I removed the first.